source/core/grids.cc

```diff
@@ -394,6 +394,99 @@ read_mesh_and_manifolds(
     }
 }
 
+template <int dim, int spacedim>
+void
+read_mesh_and_manifolds_for_stator_and_rotor(
+  parallel::DistributedTriangulationBase<dim, spacedim> &triangulation,
+  const Parameters::Mesh                                &mesh_parameters,
+  const bool                                            &restart,
+  const Parameters::Mortar<dim>                         &mortar_parameters)
+{
+  AssertThrow(
+    mesh_parameters.type == Parameters::Mesh::Type::dealii,
+    ExcMessage(
+      "GMSH files are currently not compatible with the mortar implementation."));
+
+  // Stator triangulation
+  Triangulation<dim> stator_temp_tria;
+  attach_grid_to_triangulation(stator_temp_tria, mesh_parameters);
+
+  // Rotor triangulation
+  Triangulation<dim> rotor_temp_tria;
+  attach_grid_to_triangulation(rotor_temp_tria, *mortar_parameters.rotor_mesh);
+  GridTools::rotate(mortar_parameters.rotor_mesh->rotation_angle,
+                    rotor_temp_tria);
+
+  // Check manifold IDs at rotor boundary
+  std::set<unsigned int> rotor_boundary_manifold_ids;
```

Comment:
I must admit I am not 100% sure if this function is doing what it should do.

First of all the question who is responsible for shifting boundary/manifold indices!? Let's discuss this issue in person, since it will be relevant once we want to specify via parameter files which face should be coupled to which.

Second, why do we only consider boundary manifolds? Manifolds can also be attached to cells and edges and these would be lost with the current implementation.

Generally, we could consider to extend `GridGenerator::merge_triangulations()` in deal.II.